app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,64 @@
+class OrdersController < ApplicationController
+  def index
+    @orders = Order.for_user(current_user)
+  end
+
+  def show
+    @order = Order.find(params[:id])
+  end
+
+  def no_cart
+    flash[:alert] = 'you have not put anything in the cart'
+    redirect_to root_path
+  end
+
+  def create
+    if current_user.present?
+      order = find_order
+      if order.blank?
+        params[:alert] = "This Order is not belongs to you"
+        redirect_to :back
+        return
+      end
+
+      if !params[:order_detail][:quantity].to_s.match(/\A[+-]?\d+?(\.\d+)?\Z/) || params[:order_detail][:quantity].to_i < 0
+        flash[:alert] = 'quantity must be number and greater than 0'
+        redirect_to :back
+        return
+      end
+
+      book = Book.find(params[:order_detail][:book_id])
+
+      order_detail = OrderDetail.find_by(book: book, order: order)
+      success = false
+      if order_detail.present?
+        order_detail.quantity += params[:order_detail][:quantity].to_i
+        success = order_detail.save
+      else
+        order_detail = OrderDetail.create(book: book, quantity: params[:order_detail][:quantity], order: order)
```

Comment:
if code is over 80 characters, it'd be hard to put it in a book, as demonstrated by wrapping here. Maybe this:

`OrderDetail.create(
  book: book,
  quantity: params[:order_detail][:quantity],
  order: order,
)`